src/main/java/decodex/Decodex.java

```diff
@@ -77,9 +78,7 @@ public void run() {
             } catch (RecipeManagerException e) {
                 editingRecipeName = null;
             }
-            ui.printPromptHeader(editingRecipeName);
-            String userInput = ui.readInput();
-            logger.fine("User input: " + userInput);
+            String userInput = promptAndGetUserInput(editingRecipeName);
```

Comment:
Perhaps undo this change as this could result in the redrawing of the PUML diagrams?